test/TensorFlow/device_placement.swift

```diff
@@ -27,6 +27,16 @@ public func explicitDeviceConfigGPU() {
 // CHECK-LABEL: --- TFPartition Accelerator Result: {{.*}}explicitDeviceConfigGPU{{.*}}
 // CHECK: graph_op "Const"() {dtype$dtype: $Builtin.FPIEEE32, value$tensor: f32 0x3F800000 /* 1 */, __device: "ALL_DEVICES"
 
+public func closureDeviceConfigGPU() {
+  withGPU {
+    let x = Tensor<Float>(1.0)
+    _hostOp(x)
+  }
+}
+
+// CHECK-LABEL: --- TFPartition Accelerator Result: {{.*}}closureDeviceConfigGPU{{.*}}
+// CHECK: graph_op "Const"() {dtype$dtype: $Builtin.FPIEEE32, value$tensor: f32 0x3F800000 /* 1 */, __device: "ALL_DEVICES"
```

Comment:
scalar promotion always gets onto all devices.

should we extend the test case with something like an add within the closure, and check it's on GPU? Also add some tensor code after the closure, and confirm that's on CPU?